plugins/proofpoint_tap/komand_proofpoint_tap/tasks/monitor_events/task.py

```diff
@@ -114,28 +127,35 @@ def run(self, params={}, state={}):  # noqa: MC0001
                 state[self.PREVIOUS_LOGS_HASHES] = []
                 return [], state, False, error.status_code, error
         except Exception as error:
-            self.logger.info(f"Exception occurred in monitor events task: {error}")
+            self.logger.info(f"Exception occurred in monitor events task: {error}", exc_info=True)
             state[self.PREVIOUS_LOGS_HASHES] = []
             return [], state, has_more_pages, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
 
     @staticmethod
     def get_current_time():
         return datetime.now(timezone.utc)
 
-    def parse_logs(self, unparsed_logs: list) -> list:
+    def parse_logs(self, unparsed_logs: dict) -> list:
         parsed_logs = []
         for event_type in ["clicksBlocked", "clicksPermitted", "messagesBlocked", "messagesDelivered"]:
             parsed_logs.extend(self.prepare_log(log, event_type) for log in unparsed_logs.get(event_type, []))
         return parsed_logs
 
-    @staticmethod
-    def prepare_log(log: dict, value: str) -> dict:
+    def prepare_log(self, log: dict, value: str) -> dict:
         log["eventType"] = value
 
         # preventing random sorting of the list to ensure that the same hash is generated with each request
-        if log.get("messageParts"):
-            log["messageParts"] = sorted(log.get("messageParts", []), key=lambda part: part.get("md5", ""))
-        return dict(sorted(log.items()))
+        try:
```

Comment:
Previously if we hit the error on apply `sort` against a None value and then eventually hit the logic around `max_allowed_lookback` and move the time stamps forward. This now won't be hit with the backfill logic, so we've added the except to log this error and skip this log.